### ISSUE TITLE
Bug with @ key on german keyboard

### ISSUE THREAD

**Issue by Mempler** [ppy/osu-framework#1965]
I've just noticed that when you're pressing the key `@` (`alt gr + q`) you also have to press another key to get the `@` character.

![Proof](https://onii-chan-please.come-inside.me/2018-10-29_20-10-15.gif)

(Tested on latest git clone)

[Runtime Log](https://onii-chan-please.come-inside.me/runtime.log)

Catched on osu client.

**Comment by MrTheMake** [comment]
Is this still relevant? I can't reproduce this issue.

**Comment by Gabixel** [comment]
Still happens for me with Italian keyboard (and my problem is the same of this duplicated issue: ppy/osu/issues/3884)

**Comment by bdach** [comment]
I decided to investigate this a little bit. Not only do the AltGr combination keys not work, [Alt codes](https://en.wikipedia.org/wiki/Alt_code) don't either. It's an esoteric input method, I know.

After a short browse I found these lines:
https://github.com/ppy/osu-framework/blob/7d8df261cc21023dd014444fd5acbf23671241b3/osu.Framework/Graphics/UserInterface/TextBox.cs#L712-L713
Root cause seems to be that those conditions for `e.AltPressed` *and* `e.ControlPressed` (because surprise, AltGr is secretly Ctrl+Alt, at least on Windows) prevent character input using those keys from being consumed. Removing them reenables both AltGr and Alt codes.

Last change to lines above was in 6ebcbb97. I'm lacking context here so it's hard for me to say what the correct fix would be here. I might prop up a tiny PR for this, though I had trouble getting text boxes in visual tests to consume keyboard input through the `InputManager`, so I resorted to more primitive tactics to diagnose. I would want to submit a test for this, though, just for good practice.
